cva6/tests/custom/sv32/vm_misaligned_superpage.S

```diff
@@ -0,0 +1,207 @@
+#=======================================================================
+#  Misaligned Superpage for level 1 PTE for supevisor and user mode
+#-----------------------------------------------------------------------
+# Test Description:
+#
+# If PTE at level1 is leaf PTE (superpage) and its pte.ppn[0]!=0, then
+# it is a misaligned superpage and accessing that PTE would raise page
+# fault exception of the corresponding access type.
+# 
+# When satp.mode=sv32, PTE has (r,w,x) PMP permissions, PTE has 
+# non-reserved RWX encodings, (pte.r | pte.x)=1, and pte.v=1, then test
+# the following in supervisor and user privilege mode for level1 PTE.
+# 
+# - set pte.ppn[0]!=0 and test for read, write and execute access.
+#
+#=======================================================================
+
+#include "macros.h"
+
+#ifdef smode
+    #define SET_PTE_U 0
+#else
+    #define SET_PTE_U PTE_U
+#endif
+
+#define _MMODE_  "M"
+#define _SUMODE_ "SU"
+
+.text
+.global _start
+
+_start:
+
+    ALL_MEM_PMP                                                            # PMP permission to all the memory
+    la t1,trap_handler                                                     # loads the address of trap handler 
+    csrw mtvec,t1                                                          # sets the mtvec to trap handler 
+
+# ----------------LEVEL 1 PTE Setup for load and store test------------
+
+    la a1,vm_en                                                            # loads the address of label vm_en
+    mv a0, a1                                                              # set the VA to PA (identity mapping)
+    ori a2, x0, ( PTE_D | PTE_A | SET_PTE_U | PTE_X | PTE_V )              # sets the permission bits
+    PTE_SETUP_RV32 ( a1, a2, t1, a0, pgtb_l1, LEVEL1 )                     # setup the PTE for level1
+ 
+    la a1,rvtest_data                                                      # loads the address of label rvtest_data
```

Comment:
how are making sure that `pte.ppn[0]!=0` in this test?

Reply:
Regarding the load and store test, please note that the address of the `rvtest_data` label has `ppn[0]!=0`, and this is achieved through the use of .`align 22 `on line **191.**

However, in the context of the execute test, it's important to highlight that the address of the `check_execute` label also follows the condition `ppn[0]!=0,` and this alignment is established using `.align 12` on line **116**